lib/cli/run.js

```diff
@@ -290,8 +289,5 @@ exports.builder = yargs =>
 exports.handler = argv => {
   debug('post-yargs config', argv);
   const mocha = new Mocha(argv);
-  const files = handleFiles(argv);
-
-  debug('running tests with files', files);
-  runMocha(mocha, argv, files);
+  runMocha(mocha, argv);
```

Comment:
This is existing code, I know. But `argv` is passed twice to `runMocha()`, once with `mocha` instance and a second time as a function parameter. Some options may have been processed meanwhile within `mocha`.

Reply:
Do you have a suggestion how to improve this?

Reply:
At least `grep` is processed within the `Mocha` constructor, therefore using the unprocessed `grep` of `argv` seems incorrect to me. Why don't we use just `mocha.options`?

I also wonder why I don't see `fgrep` anywhere. It is also processed within the `Mocha` constructor, but not used in `runMocha()`.

Reply:
Is it ok if I address this separately? It seems to be an orthogonal concern to me and there is a chance this changes the behavior, something I’d like to avoid. It also looks like this should be tackled more comprehensively by rethinking whether `handler` and `runMocha` should be separate functions.

Reply:
I started to address this in https://github.com/mochajs/mocha/pull/3960 which should also answer the question about `fgrep`.